### PR TITLE
add doc for p4c compiler driver

### PR DESCRIPTION
**Pull request by hanw**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -252,3 +253,73 @@ output:
 
   * use `ordered_map` and `ordered_set` when you need to iterate;
     they provide deterministic iterators
+
+## Compiler Driver
+
+**p4c** is a compiler driver. The goal is to provide a cnosistent user interface
+across different p4 backends and work flows. The compiler driver is written in 
+Python. It can be extended for custom backend.
```
Comment (cc10512) [inline docs/README.md]: custom backends.

File: docs/README.md [hunk 2]
```diff
@@ -252,3 +253,73 @@ output:
...
+
+The usage of the driver is as follows:
+```
+usage: p4c [-h] [-V] [-v] [-###] [-Xpreprocessor <arg>] [-Xp4c <arg>]
+           [-Xassembler <arg>] [-Xlinker <arg>] [-b BACKEND] [-E] [-e] [-S]
+           [-c] [-x {p4-14,p4-16}] [-I SEARCH_PATH] [-o PATH] [--target-help]
+           [source_file]
+
+positional arguments:
+  source_file           Files to compile
```
Comment (cc10512) [inline docs/README.md]: can we actually compile multiple files or just one. AFAIK, one only.

File: docs/README.md [hunk 3]
```diff
@@ -252,3 +253,73 @@ output:
...
+```
+# In your custom Makefile.am
+
+p4c_PYTHON += p4c.custom.cfg
+```
+
+There is an global variable `config` in p4c compiler driver that stores the build steps 
+for a particular target. By default, the bmv2 and ebpf backends are supported. Each backend
+is identified with a triplet, target-arch-vendor. For example, the default bmv2 backend is 
+identified as bmv2-*-p4org. The * is a wildcard to represent any architecture. User may choose
```
Comment (cc10512) [inline docs/README.md]: might want to quote `bmv2-*-p4org`

File: docs/README.md (hunk 2 above)
Comment (hanw) [inline docs/README.md]: for now, only one.

**Pull request merged**